include/envoy/router/router.h

```diff
@@ -370,6 +370,22 @@ class ShadowPolicy {
 
 using ShadowPolicyPtr = std::unique_ptr<ShadowPolicy>;
 
+/**
+ * All virtual cluster stats. @see stats_macro.h
+ */
+#define ALL_VIRTUAL_CLUSTER_STATS(COUNTER)                                                         \
+  COUNTER(upstream_rq_retry)                                                                       \
+  COUNTER(upstream_rq_retry_success)                                                               \
+  COUNTER(upstream_rq_timeout)                                                                     \
+  COUNTER(upstream_rq_total)
```

Comment:
I was mulling over defining another layer of macros so you could pre-declare these names in a config structure or factory, without binding to a scope, and then have a separate macro that instantiates the stats in a scope.